### PR TITLE
Fix missing `power_state` attribute value for `azurerm_virtual_machine` and `azurerm_virtual_machine_scale_set` data sources

### LINKED ISSUE #26990: Power State not reported in azurerm_virtual_machine and azurerm_virtual_machine_scale_set data sources
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.9.3

### AzureRM Provider Version

3.115.0

### Affected Resource(s)/Data Source(s)

azurerm_virtual_machine azurerm_virtual_machine_scale_set 

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3.0"
    }
  }
}

provider "azurerm" {
  features {}
}

data "azurerm_virtual_machine" "example" {
  name                = "check-vm"
  resource_group_name = "check-vm"
}

output "power_state" {
  value = data.azurerm_virtual_machine.example.power_state
}

data "azurerm_virtual_machine_scale_set" "flexible" {
  name                = "vmss-test"
  resource_group_name = "vmss-test"
}

data "azurerm_virtual_machine_scale_set" "uniform" {
  name                = "uniform-test"
  resource_group_name = "vmss-test"
}

output "flexible_power" {
  value = { for instance in data.azurerm_virtual_machine_scale_set.flexible.instances : instance.name => instance.power_state }
}

output "uniform_power" {
  value = { for instance in data.azurerm_virtual_machine_scale_set.uniform.instances : instance.name => instance.power_state }
}
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

```bash
flexible_power = {
  "vmss-test_3b9d9442" = "running"
  "vmss-test_931fa536" = "running"
}
power_state = "running"
uniform_power = {
  "uniform-test_0" = "running"
  "uniform-test_1" = "running"
}
```

### Actual Behaviour

```bash
flexible_power = {
  "vmss-test_3b9d9442" = ""
  "vmss-test_931fa536" = ""
}
power_state = ""
uniform_power = {
  "uniform-test_0" = ""
  "uniform-test_1" = ""
}
```

### Steps to Reproduce

1. Provision an Azure VM, Azure VM Scale Set in Uniform mode, and VM Scale Set in Flexible mode
1. Run `terraform apply` with the supplied configuration and the correct resource group name and resource names
1. The output for the `power_state` attribute will be blank

### Important Factoids

_No response_

### References

This was previously resolved in PR #22851. Refactoring of the code to support the newer API broke it again.

### PR DESCRIPTION
**Pull request by ned1313**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

This PR fixes the problems identified in Issue #26990.

Currently, both the `azurerm_virtual_machine` and `azurerm_virtual_machine_scale_set` do not correctly report back the `power_state` attribute. The value shows up as `null`.

Additionally, the `azurerm_virtual_machine_scale_set` doesn't populate the `computer_name` attribute for scale sets running in Flexible orchestration mode.

## PR Checklist

- [X] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [X] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [X] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [X] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [X] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [X] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [X] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [X] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.

## Testing 

- [X] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

```bash
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestAccDataSourceAzureRMVirtualMachine_basicLinux$ github.com/hashicorp/terraform-provider-azurerm/internal/services/compute

ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/compute	0.024s

Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestAccDataSourceAzureRMVirtualMachine_basicWindows$ github.com/hashicorp/terraform-provider-azurerm/internal/services/compute

ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/compute	0.026s

Running tool: /usr/local/go/bin/go test -timeout 30s -run ^TestAccDataSourceVirtualMachineScaleSet_basicLinux$ github.com/hashicorp/terraform-provider-azurerm/internal/services/compute

ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/compute	0.022s
```

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_virtual_machine` data source - `power_state` attribute no longer `null` [GH-26990]
* `azurerm_virtual_machine_scale_set` data source - `power_state` and `computer_name` attribute no longer `null` [GH-26990]

This is a (please select all that apply):

- [X] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #26990

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/compute/virtual_machine_scale_set_data_source.go [hunk 1]
```diff
@@ -275,29 +296,53 @@ func getVirtualMachineScaleSetVMConnectionInfo(ctx context.Context, networkInter
...
 	output := make(map[string]interface{})
 	output["name"] = *input.Name
 	output["instance_id"] = *input.InstanceId
 
-	if props := input.Properties; props != nil {
-		if props.LatestModelApplied != nil {
-			output["latest_model_applied"] = *props.LatestModelApplied
-		}
+	if mode == "Flexible" {
+		if props := vm.Properties; props != nil {
```
Comment (stephybun) [inline internal/services/compute/virtual_machine_scale_set_data_source.go:305]: We should nil check `vm` here as well.

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @ned1313. Could you take a look at the suggestions left in-line? Once those are resolved I can kick off the tests.

File: internal/services/compute/virtual_machine_scale_set_data_source.go [hunk 2]
```diff
@@ -170,12 +170,32 @@ func dataSourceVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta interf
...
+		}
+	}
+
+	result, err := instancesClient.ListComplete(ctx, virtualMachineScaleSetId, optionsVMSS)
 	if err != nil {
 		return fmt.Errorf("listing VM Instances for %q: %+v", id, err)
 	}
 
 	var connInfo *connectionInfo
+	var vmModel virtualmachines.VirtualMachine
```
[the remark this answers is not in the document]
Comment (ned1313) [reply internal/services/compute/virtual_machine_scale_set_data_source.go]: Done!

File: internal/services/compute/virtual_machine_scale_set_data_source.go [hunk 3]
```diff
@@ -187,20 +207,21 @@ func dataSourceVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta interf
...
 				// Network Interfaces of VM in Flexible VMSS are accessed from single VM
 				virtualMachineId := virtualmachines.NewVirtualMachineID(subscriptionId, id.ResourceGroupName, *item.InstanceId)
-				vm, err := virtualMachinesClient.Get(ctx, virtualMachineId, virtualmachines.DefaultGetOperationOptions())
+				vm, err := virtualMachinesClient.Get(ctx, virtualMachineId, optionsVM)
 				if err != nil {
 					return fmt.Errorf("retrieving VM Instance %q for %q: %+v", *item.InstanceId, id, err)
 				}
 				connInfoRaw := retrieveConnectionInformation(ctx, networkInterfacesClient, publicIPAddressesClient, vm.Model.Properties)
 				connInfo = &connInfoRaw
+				vmModel = *vm.Model
```
[the remark this answers is not in the document]
Comment (ned1313) [reply internal/services/compute/virtual_machine_scale_set_data_source.go]: Done!

File: internal/services/compute/virtual_machine_scale_set_data_source.go [hunk 4]
```diff
@@ -187,20 +207,21 @@ func dataSourceVirtualMachineScaleSetRead(d *pluginsdk.ResourceData, meta interf
...
 			} else {
 				connInfo, err = getVirtualMachineScaleSetVMConnectionInfo(ctx, nics.Items, id.ResourceGroupName, id.VirtualMachineScaleSetName, *item.InstanceId, vmssPublicIpAddressesClient)
 				if err != nil {
 					return err
 				}
 			}
 
-			flattenedInstances := flattenVirtualMachineScaleSetVM(item, connInfo)
+			flattenedInstances := flattenVirtualMachineScaleSetVM(item, connInfo, &vmModel, orchestrationMode)
```
[the remark this answers is not in the document]
Comment (ned1313) [reply internal/services/compute/virtual_machine_scale_set_data_source.go]: Done!

File: internal/services/compute/virtual_machine_scale_set_data_source.go (hunk 1 above)
Comment (ned1313) [reply internal/services/compute/virtual_machine_scale_set_data_source.go:305]: Added nil check to `if mode == "Flexible"`

**Review by stephybun — APPROVED** [review]
Tests looks good, thanks @ned1313 LGTM 👍

**Pull request merged**